File: Clase 1/Programas/ej_1.c
#include <stdio.h>

int main() {

int num,cont=0,acum=0,primero,cont1=0,i,contdivi=0,contprimos=0;
	printf("Ingrese un numero entero");
	scanf("%d",&num);
	primero=num;
	while(num!=0){
		if(num%6==0 && num>0){
			cont=cont+1;
			acum=acum+num;
		}
		if(num%primero==0 && num!=primero){
			cont1++;
		}
        for(i=2;i<num;i++){
			if(num%i==0){
				contdivi++;
			}
		}
			if (contdivi==0 && num!=1 && num>0){
			contprimos++;
			//printf("el numero primo es %d",num);
		    }
		contdivi=0;
	    printf("Ingrese un numero entero");
	    scanf("%d",&num);
	}
	if(cont!=0){
		printf("\nel promedio es %d",acum/cont);
	}
	else {
		printf("no se ingresaron valores para calcular el promedio");
	}
	printf("\nla cantidad de multiplos es %d %d",primero,cont1);
	printf("\nla cantidad de numeros primos es %d",contprimos);

    return 0;
}

File: Clase 1/Programas/ej_1.py


preg_num = int(input("Número: "))

cant_num = 0

suma =  0

cant_num_mult_prim_num = 0

cant_primos = 0

divisor = 2

cant_prom = 0

while preg_num != 0:

    cant_num += 1

    if (preg_num > 0 and preg_num % 6 == 0):
        suma += preg_num
        cant_prom += 1

    if cant_num == 1:

        for i in range(1, preg_num+1):

            if preg_num % i == 0:
                cant_num_mult_prim_num += 1


    while preg_num > 1 and preg_num % divisor != 0 and preg_num != divisor:
        divisor += 1

    if divisor == preg_num:
        cant_primos += 1

    
    preg_num = int(input("Número: "))


if cant_prom > 0:
    promedio = suma / cant_prom
else:
    promedio = ""


print(f"Promedio: {promedio}")
print(f"Cantidad de numeros multiplos de primer numero: {cant_num_mult_prim_num}")
print(f"Cantidad de numeros primos {cant_primos}")



            
    
    